components/intercom/conversation-closed.js

```diff
@@ -0,0 +1,54 @@
+const intercom = require("https://github.com/PipedreamHQ/pipedream/components/intercom/intercom.app.js");
+
+module.exports = {
+  name: "New Closed Conversation",
+  description: "Emits an event each time a conversation is closed.",
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
+    intercom,
+    db: "$.service.db",
+    timer: {
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 15,
+      },
+    },
+  },
+
+  async run(event) {
+    const monthAgo = this.intercom.monthAgo();
+    let lastConversationClosedAt =
+      this.db.get("lastConversationClosedAt") || Math.floor(monthAgo / 1000);
+    const data = {
+      query: {
+        field: "statistics.last_close_at",
+        operator: ">",
+        value: lastConversationClosedAt,
+      },
+    };
+
+    let results = null;
+    let starting_after = null;
+
+    while (
+      !results ||
+      (results.data.pages.next !== null &&
```

Comment:
Do you care about the case where `results.data.pages.next === ""` (empty string)? Otherwise you can just check for `results.data.pages.next` being truthy:

```
while (results || results.data.pages.next) {
```
```